include/inviwo/core/datastructures/bitset.h

```diff
@@ -0,0 +1,399 @@
+
+    void clear();
+
+    /**
+     * check whether this bitset is a subset of \p b
+     */
+    bool isSubsetOf(const BitSet& b) const;
+    /**
+     * check whether this bitset is a strict subset of \p b
+     */
+    bool isStrictSubsetOf(const BitSet& b) const;
+
+    /**
+     * Add value \p v to the bitset
+     *
+     * @param v  value
+     */
+    void add(uint32_t v);
+
+    void add(const std::vector<uint32_t>& values);
+
+    template <size_t N>
+    void add(const std::array<uint32_t, N>& values) {
+        addMany(values.size(), values.data());
+    }
+
+    template <typename InputIt,
+              class = typename std::enable_if_t<detail::is_iterator<InputIt>::value>>
+    void add(InputIt begin, InputIt end) {
+        while (begin != end) {
+            add(*begin);
+            ++begin;
+        }
+    }
+
+    template <typename... Ts>
+    void add(uint32_t val, Ts&&... values) {
+        add(val);
+        add(values...);
+    }
+
+    /**
+     * Add value \p v to the bitset and return true on success
+     *
+     * @param v  value
+     * @return true if \p v was added, false if it already existed in the bitset
+     */
+    bool addChecked(uint32_t v);
+
+    /**
+     * adds all values from the open range [min,max)
+     *
+     * @param min   lower bound
+     * @param max   upper bound (not included)
+     */
+    void addRange(uint32_t min, uint32_t max);
+
+    /**
+     * adds all values from the closed range [min,max]
+     *
+     * @param min   lower bound
+     * @param max   upper bound (included)
+     */
+    void addRangeClosed(uint32_t min, uint32_t max);
+
+    /**
+     * Remove value \p v to the bitset
+     *
+     * @param v  value
+     */
+    void remove(uint32_t v);
+
+    /**
+     * Remove value \p v to the bitset and return true on success
+     *
+     * @param v  value
+     * @return true if \p v was removed, false if it did not exist in the bitset
+     */
+    bool removeChecked(uint32_t v);
+
+    /**
+     * Return the largest value
+     *
+     * @return largest value in the bitset, 0 if empty
+     */
+    uint32_t max() const;
+    /**
+     * Return the smallest value
+     *
+     * @return smallest value in the bitset, UINT32_MAX if empty
+     */
+    uint32_t min() const;
+
+    /**
+     * Checks whether \p v is part of the bitset
+     *
+     * @param v  value to check
+     * @return true if \p v exists
+     */
+    bool contains(uint32_t v) const;
+
+    /**
+     * Checks whether the open range [\p min, \p max) is part of the bitset
+     *
+     * @param min  lower bound
+     * @param max  upper bound
+     * @return true if [\p min, \p max) exists
+     */
+    bool containsRange(uint32_t min, uint32_t max) const;
+
+    /**
+     * Flip all bits in the open range [\p min, \p max)
+     */
+    void flipRange(uint32_t min, uint32_t max);
+
+    /**
+     * Return the number of values that are less or equal to \p v
+     */
+    size_t rank(uint32_t v) const;
```

Comment:
could one use this for a faster std::distance(it1, it2) -> rank(*it2) - rank(*it1) or something like that?

Reply:
There is some internal function doing the computation
```c++
uint64_t Roaring::rank(uint32_t x) const {
    return api::roaring_bitmap_rank(&roaring, x);
}
```

Reply:
Potentially yes.